From aa9110d6abb3628f829ee7027b2cbd5c93c9da7c Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Tue, 25 May 2021 12:00:45 +0200
Subject: [PATCH] imx8mm-var-dart-nrt: Enable M4, run from TCM

Enable the M4 core, set firmware binary name,
set the DTB and let the binary reside in the TCM
(as it came by default).

But, we stay on default DTB for now, as the M4 dtb:
- disables ecspi1 (no spidev)
- reserves DDR memory for the M4
- sets up rpmsg for M4

So, we leave this patch here for when it will be
needed, but currently M4 is not enabled from dtb.

Upstream-status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 include/configs/imx8mm_var_dart.h | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/include/configs/imx8mm_var_dart.h b/include/configs/imx8mm_var_dart.h
index b00e8ba245..ad80db716a 100644
--- a/include/configs/imx8mm_var_dart.h
+++ b/include/configs/imx8mm_var_dart.h
@@ -79,10 +79,9 @@
 	"mmcautodetect=yes\0" \
 	"mmcpart=1\0" \
 	"m4_addr=0x7e0000\0" \
-	"m4_bin=hello_world.bin\0" \
-	"use_m4=no\0" \
-	"loadm4bin=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bootdir}/${m4_bin}; " \
-		"cp.b ${loadaddr} ${m4_addr} ${filesize}\0" \
++       "m4_bin=m4_fw.bin\0" \
++       "use_m4=yes\0" \
++       "loadm4bin=load mmc ${mmcdev}:${resin_boot_part} ${m4_addr} ${m4_bin}\0" \
 	"runm4bin=" \
 		"if test ${m4_addr} = 0x7e0000; then " \
 			"echo Booting M4 from TCM; " \
-- 
2.17.1

